### ISSUE TITLE
CLI action chaining

### ISSUE THREAD

**Issue by admk** [Aaron-Zhao123/mayo-dev#15]
Currently, the CLI has the following interface:
```
❯ ./my
Usage:
    my eval [<anything>...]
    my export [<anything>...]
    my info [<anything>...]
    my train [<anything>...]
    my (-h | --help)
```

We now consider changing the CLI to the following:
```
❯ ./my
Usage:
    my <action>... [<anything>...]
    my (-h | --help)

Available Actions:
    eval export info train train_once update_overrider save_checkpoint
```
So that the CLI allows easily manual chaining of a sequence of actions to be performed, e.g.:
```
❯ ./my export info eval update_overrider once once once save_checkpoint \
    <yaml files>...
```
The above command will run the following tasks in sequence:
* Export YAML
* Display graph info
* Evaluates validation accuracy
* Updates overrider variables
* Execute training step three times
* Save checkpoint

**Comment by admk** [comment]
I envision another possible unified interface to chain everything together:
```
Usage:
    ./my <anything>...

<anything> can be one of the following:
    - action, one of: eval export info train update_overrider;
    - a YAML file; or
    - an overrider argument, e.g., system.num_gpus=2.
```

So it would allow any configs and commands to be interpreted in a sequence, for instance the command:
```
❯ ./my models/mobilenet.yaml dataset/imagenet.yaml info trainers/mobilenet.yaml \
        system.num_gpus=2 system.max_steps=1000 train update_overrider \
        system.max_steps=2000 train eval
```
would perform the following actions in sequence:
1. load `models/mobilenet.yaml`;
2. load `dataset/imagenet.yaml`;
3. print out network information;
4. load `trainers/mobilenet.yaml`;
5. set number of gpus to use for training to 2;
6. set maximum number of steps to train to 1000 steps;
7. train until 1000 steps;
7. update variable overrider states;
8. set maximum number of steps to train to 2000 steps;
9. fine-tune until 2000 steps;
10. evaluate the performance of the final model.
